### PR TITLE
Fix wallet unlock 

### PR DESCRIPTION
**Pull request by prushin**

### CODE REVIEW

File: src/LndClient/RPC/Katip.hs [hunk 1]
```diff
@@ -227,14 +227,14 @@ catchWalletLock ::
 catchWalletLock env x = do
   x0 <- x
   case x0 of
-    Left LndWalletLocked -> do
-      $(logTM) (newSev env WarningS) "Wallet is locked, needs to be unlocked"
+    Left err -> do
```
Comment (prushin) [inline src/LndClient/RPC/Katip.hs:230]: There is no specific error text, also use specific text is not relaible, so let's try to intercept any error and if it helps, okay, if not - function will return error.

**Review by drownbes — APPROVED** [review]

File: src/LndClient/RPC/Katip.hs [hunk 2]
```diff
@@ -228,13 +228,13 @@ catchWalletLock env x = do
...
     Left LndWalletLocked -> do
-      $(logTM) (newSev env WarningS) "Wallet is locked, needs to be unlocked"
+      $(logTM) (newSev env WarningS) "Wallet possibly is locked, try to unlock"
       unlocked <- lazyUnlockWallet env
       case unlocked of
         Right () -> do
           $(logTM) (newSev env InfoS) "Wallet unlocked"
-          pure x0
-        Left err -> do
+          x
```
Comment (tim2CF) [inline src/LndClient/RPC/Katip.hs:236]: This is implicit repeat of original RPC call. It's dangerous, some RPCs might be not suitable for safe retries. For example SendCoins. Let's leave it as it was, just returning result whatever is was. Retries are handled separately in other combinator.

File: src/LndClient/RPC/Katip.hs [hunk 3]
```diff
@@ -228,13 +228,13 @@ catchWalletLock env x = do
...
+        Left err0 -> do
           $(logTM) (newSev env ErrorS) "Wallet unlock failure"
-          pure $ Left err
+          pure $ Left err0
```
Comment (tim2CF) [inline src/LndClient/RPC/Katip.hs:239]: The same, unlock is just a side effect, let's have original result here.

File: src/LndClient/RPC/Silent.hs [hunk 4]
```diff
@@ -203,6 +203,6 @@ catchWalletLock env x = do
     Left LndWalletLocked -> do
       unlocked <- lazyUnlockWallet env
       case unlocked of
-        Right () -> pure x0
+        Right () -> x
```
Comment (tim2CF) [inline src/LndClient/RPC/Silent.hs:206]: The same, do not retry here, leave it as it was.

File: src/LndGrpc/Client.hs [hunk 5]
```diff
@@ -36,14 +36,13 @@ runUnary rpc env req = do
   res <-
     catchExceptions $ liftIO (Right <$> this)
   pure $ case res of
-    Right (Right (Right (Right (_, _, Right x)))) ->
+    Right (Right (Right (Right (_, _, Right x)))) -> do
```
Comment (tim2CF) [inline src/LndGrpc/Client.hs:39]: "do" is redundant

**Review by tim2CF — CHANGES_REQUESTED** [review]
Some fixes

**Review by tim2CF — APPROVED** [review]
Good job!

**Pull request merged**